src/parentheses/Solution.java

```diff
@@ -0,0 +1,50 @@
+package parentheses;
+
+import java.util.Stack;
+
+public class Solution {
+    public boolean isValid(String parentheses) {
+
+        char[] arrayParentheses = parentheses.toCharArray();
+        Stack<Character> stack = new Stack<>();
+        boolean allOk = true;
+        for (int i = 0; i < arrayParentheses.length; i++) {
+            char character = arrayParentheses[i];
+            boolean isOpencharacter = character == '[' || character == '{' || character == '(';
+
+            if (isOpencharacter) {
+                stack.push(character);
+                System.out.println("stack: " + stack);
+            } else {
+                switch (character) {
+                    case ')':
+                        allOk = checkCloseParentheses(stack, '(');
+                        break;
+                    case '}':
+                        allOk = checkCloseParentheses(stack, '{');
+                        break;
+                    case ']':
+                        allOk = checkCloseParentheses(stack, '[');
+                        break;
+                    default:
+                        throw new IllegalStateException("Unexpected value: " + arrayParentheses[i]);
```

Comment:
Why use `arrayParentheses[i]` instead of the variable `character`?

Reply:
I used arrayParentheses[i] because it contains the character that causes the exception.

Reply:
Ok, I can use the variable "character" which also contains the information that causes the exception.